src/core/infer_request.cc

```diff
@@ -590,14 +590,17 @@ InferenceRequest::Normalize()
     }
   }
 
-  // Make sure that the request is providing the same number of inputs
+  // Make sure that the request is providing the number of inputs
   // as is expected by the model.
-  if (original_inputs_.size() != (size_t)model_config.input_size()) {
+  if ((original_inputs_.size() > (size_t)model_config.input_size()) ||
+      (original_inputs_.size() < model_raw_->RequiredInputCount())) {
     return Status(
         Status::Code::INVALID_ARG,
-        "expected " + std::to_string(model_config.input_size()) +
-            " inputs but got " + std::to_string(original_inputs_.size()) +
-            " inputs for model '" + ModelName() + "'");
+        "expected number of inputs between [" +
+            std::to_string(model_raw_->RequiredInputCount()) + ", " +
```

Comment:
I would not use [ ], just say "... inputs between # and # but got # inputs for model"